spring-cloud-gateway-server-mvc/src/test/java/org/springframework/cloud/gateway/server/mvc/handler/ProxyExchangeHandlerFunctionTest.java

```diff
@@ -0,0 +1,142 @@
+import org.springframework.cloud.gateway.server.mvc.config.GatewayMvcProperties;
+import org.springframework.cloud.gateway.server.mvc.filter.HttpHeadersFilter.RequestHttpHeadersFilter;
+import org.springframework.cloud.gateway.server.mvc.filter.HttpHeadersFilter.ResponseHttpHeadersFilter;
+import org.springframework.http.HttpHeaders;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.web.servlet.function.ServerRequest;
+import org.springframework.web.servlet.function.ServerResponse;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author raccoonback
+ */
+class ProxyExchangeHandlerFunctionTest {
+
+	@Test
+	void keepOriginalEncodingOfQueryParameter() {
+		TestProxyExchange proxyExchange = new TestProxyExchange();
+		ProxyExchangeHandlerFunction function = new ProxyExchangeHandlerFunction(proxyExchange, new ObjectProvider<>() {
+			@Override
+			public RequestHttpHeadersFilter getObject() throws BeansException {
+				return null;
+			}
+
+			@Override
+			public RequestHttpHeadersFilter getObject(Object... args) throws BeansException {
+				return null;
+			}
+
+			@Override
+			public RequestHttpHeadersFilter getIfAvailable() throws BeansException {
+				return null;
+			}
+
+			@Override
+			public RequestHttpHeadersFilter getIfUnique() throws BeansException {
+				return null;
+			}
+
+			@Override
+			public Stream<RequestHttpHeadersFilter> orderedStream() {
+				return Stream.of((httpHeaders, serverRequest) -> new HttpHeaders());
+			}
+
+		}, new ObjectProvider<>() {
+
+			@Override
+			public ResponseHttpHeadersFilter getObject() throws BeansException {
+				return null;
+			}
+
+			@Override
+			public ResponseHttpHeadersFilter getObject(Object... args) throws BeansException {
+				return null;
+			}
+
+			@Override
+			public ResponseHttpHeadersFilter getIfAvailable() throws BeansException {
+				return null;
+			}
+
+			@Override
+			public ResponseHttpHeadersFilter getIfUnique() throws BeansException {
+				return null;
+			}
+
+			@Override
+			public Stream<ResponseHttpHeadersFilter> orderedStream() {
+				return Stream.of((httpHeaders, serverRequest) -> new HttpHeaders());
+
+			}
+		});
+
+		function.onApplicationEvent(null);
+
+		MockHttpServletRequest servletRequest = MockMvcRequestBuilders
+			.get("http://localhost/é?foo=value1 value2&bar=value3=")
+			.buildRequest(null);
+		servletRequest.setAttribute(MvcUtils.GATEWAY_REQUEST_URL_ATTR, URI.create("http://localhost:8080"));
+		ServerRequest request = ServerRequest.create(servletRequest, Collections.emptyList());
+
+		function.handle(request);
+
+		URI uri = proxyExchange.getRequest().getUri();
+
+		assertThat(uri).hasToString("http://localhost:8080/%C3%A9?foo=value1%20value2&bar=value3%3D")
+			.hasPath("/é")
+			.hasParameter("foo", "value1 value2")
+			.hasParameter("bar", "value3=");
```

Comment:
Please also add a case for encode plus symbol (%2B).

Reply:
@Shawyeok 
I have updated it.